crates/orchestrator/src/cron/event_bridge.rs

```diff
@@ -0,0 +1,102 @@
+use std::time::Duration;
+
+use async_trait::async_trait;
+use aws_sdk_eventbridge::types::{InputTransformer, RuleState, Target};
+use aws_sdk_sqs::types::QueueAttributeName;
+
+use crate::cron::Cron;
+use crate::queue::job_queue::WorkerTriggerType;
+use crate::setup::SetupConfig;
+
+pub struct AWSEventBridge {}
+
+const WORKER_TRIGGER_RULE_NAME: &str = "worker_trigger_scheduled";
+
+#[async_trait]
+impl Cron for AWSEventBridge {
+    #[allow(unreachable_patterns)]
+    async fn setup_cron(
+        &self,
+        config: &SetupConfig,
+        cron_time: Duration,
+        target_queue_name: String,
+        message: String,
+        worker_trigger_type: WorkerTriggerType,
+    ) -> color_eyre::Result<()> {
+        let config = match config {
+            SetupConfig::AWS(config) => config,
+            _ => panic!("Unsupported Event Bridge configuration"),
+        };
+        let event_bridge_client = aws_sdk_eventbridge::Client::new(config);
+        let sqs_client = aws_sdk_sqs::Client::new(config);
+
+        event_bridge_client
+            .put_rule()
+            .name(WORKER_TRIGGER_RULE_NAME)
+            .schedule_expression(duration_to_rate_string(cron_time))
+            .state(RuleState::Enabled)
+            .send()
+            .await?;
+        let queue_url = sqs_client.get_queue_url().queue_name(target_queue_name).send().await?;
+
+        let queue_attributes = sqs_client
+            .get_queue_attributes()
+            .queue_url(queue_url.queue_url.unwrap())
+            .attribute_names(QueueAttributeName::QueueArn)
+            .send()
+            .await?;
+        let queue_arn = queue_attributes.attributes().unwrap().get(&QueueAttributeName::QueueArn).unwrap();
+
+        // Create the EventBridge target with the input transformer
+        let input_transformer =
+            InputTransformer::builder().input_paths_map("$.time", "time").input_template(message).build()?;
+
+        event_bridge_client
+            .put_targets()
+            .rule(WORKER_TRIGGER_RULE_NAME)
+            .targets(
+                Target::builder()
+                    .id(format!("worker-trigger-target-{:?}", worker_trigger_type))
```

Comment:
do we need a `worker_trigger_type`?

Reply:
Ok so when we need to remove this trigger in future or if we want to reset the setup we would need this id. This particular type of format is easier to construct.